### PR TITLE
Make 0 sample rate a valid rate

### PR DESCRIPTION
**Pull request by romtsn**
## :scroll: Description

Allow 0.0 sample rate to be set (though the effect would be as setting sample rate to null).

## :bulb: Motivation and Context

To align with other SDKs (RN, Cocoa, Flutter)

## :green_heart: How did you test it?
Automated tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

**Review by markushi — APPROVED** [review]
LGTM! [The docs](https://docs.sentry.io/platforms/android/configuration/options/#sample-rate) already mention "a value between 0.0 and 1.0", so no need to update anything there.

**Review by adinauer — APPROVED** [review]
LGTM

File: sentry/src/test/java/io/sentry/SentryOptionsTest.kt [hunk 1]
```diff
@@ -91,11 +91,6 @@ class SentryOptionsTest {
         assertFailsWith<IllegalArgumentException> { SentryOptions().sampleRate = -0.0000000000001 }
     }
 
-    @Test
-    fun `when setSampling is set to exactly 0, setter throws`() {
```
Comment (stefanosiano) [inline sentry/src/test/java/io/sentry/SentryOptionsTest.kt:95]: maybe add a test `when setSampleRate is set to exactly 0, value is set` like it's done for `setTracesSampleRate`?

Comment (romtsn) [reply sentry/src/test/java/io/sentry/SentryOptionsTest.kt:95]: We already test this as part of `SampleRateUtilsTest`, and in the options we just use the same logic, so I'd keep it like that if you don't mind

**Pull request merged**